yolact.py

```diff
@@ -478,22 +479,36 @@ def save_checkpoint(self, path, recipe=None, epoch=0):
         checkpoint = {
                          'epoch': epoch,
                          'model_state_dict': self.state_dict(),
-                         'recipe': recipe,
+                         'checkpoint_recipe': recipe,
                      }
         torch.save(checkpoint, path)
 
-    def load_checkpoint(self, path, recipe=None, resume: bool = False) -> Tuple[Optional[int], Optional[str], SparseMLWrapper]:
+    def load_checkpoint(
+        self,
+        path: str,
+        train_recipe: Optional[str] = None,
+        resume: bool = False,
+    ) -> Tuple[Optional[int], Optional[str], SparseMLWrapper]:
         """ Loads weights into the current Yolact object.
 
-        :return: A tuple containing the epoch and the recipe if
+        :param path: local path to the checkpoint
+        :param train_recipe: A train recipe if different from checkpoint recipe
+        :param resume: True if resuming a last run
+        :return: A tuple containing the epoch and the checkpoint_recipe if
             found in the checkpoint, and a SparseML Wrapper instance
         """
 
-        checkpoint = torch.load(path)
-        epoch = checkpoint.get('epoch')
-        recipe = recipe or checkpoint.get('recipe')
-        state_dict = checkpoint.get('model_state_dict', checkpoint)
-        sparseml_wrapper = SparseMLWrapper(self, recipe)
+        loaded_checkpoint = torch.load(path)
+        checkpoint_epoch = loaded_checkpoint.get('epoch', float('inf'))
+        checkpoint_recipe = loaded_checkpoint.get('checkpoint_recipe')
+        state_dict = loaded_checkpoint.get('model_state_dict', loaded_checkpoint)
+
+        sparseml_wrapper = SparseMLWrapper(
+            model=self,
+            recipe=train_recipe,
+            checkpoint_recipe=checkpoint_recipe,
+            checkpoint_epoch=float('inf') if not resume else checkpoint_epoch,
```

Comment:
We have an uncovered cased here - you're transfer learning, your run gets interrupted, and you go to resume it. When you go to apply the recipe, the checkpoint epoch isn't going to align with the epoch in the staged recipe you want to apply. i.e. the checkpoint epoch might only fall in the range of the first stage, rather than the second

Reply:
agreed, some cases are missing that will be addressed in the upcoming stages epoch project. this also brings up another issue that the checkpoint epoch is the training epoch, not the staged one, @rahul-tuli will put up a fix for that

Reply:
Fixed staged epoch!